src/samplers/activelearner.py

```diff
@@ -0,0 +1,318 @@
+    """
+    Active Learning BMDAL Static Pool Sampler.
+
+    Loads a database of runs that have already been collected.
+    This is useful for testing various acquisition strategies
+    in Active Learning on fixed datasets. Uses Pandas under the hood.
+    """
+
+    sampler_interface = S.ACTIVEDB
+
+    def __init__(self, **kwargs):
+        """
+        Initializes the Active Learning BMDAL Static Pool Sampler.
+
+        Args:
+            data_path (str): The path where the data is stored. Accepted formats: csv, pkl, h5.
+        """
+        super().__init__(**kwargs)
+
+    def collect_batch_results(self, results: list[dict[str, dict]]) -> torch.Tensor:
+        """
+        Collects results, ignoring outputs as this is labeled static pool-based.
+
+        Args:
+            results (list): List of dictionaries containing batch results.
+
+        Returns:
+            torch.Tensor: Tensor containing batch results.
+
+        """
+        idxs_as_tensor = torch.tensor([res["input"]["pool_idxs"] for res in results])
+        return idxs_as_tensor
+
+    def get_initial_parameters(self) -> list[dict]:
+        """
+        Returns a subset of initial parameters.
+
+        Returns:
+            list: List of dictionaries containing initial parameter samples.
+
+        """
+        params = []
+        for _ in range(self.init_num_samples):
+            idxs = np.random.randint(0, len(self.parser.pool))
+            result = {
+                "input": self.parser.pool[idxs, self.parser.input_col_idxs],
+                "output": self.parser.pool[idxs, self.parser.output_col_idxs],
+                "pool_idxs": idxs,
+            }
+            params.append(result)
+        return params
+
+    def get_next_parameter(
+        self, model: torch.nn.Module, train: torch.Tensor, pool: torch.Tensor
+    ) -> list[dict[str, float]]:
+        """
+        Generates the next parameter samples based on the model predictions.
+
+        Args:
+            model (torch.nn.Module): PyTorch model.
+
+        Returns:
+            list: List of dictionaries containing next parameter samples.
+
+        """
+        idxs = self._get_new_idxs_from_pool(model, train, pool)
+        results = []
+        for idx in idxs:
+            results.append(
+                {
+                    "input": self.parser.pool[idx, self.parser.input_col_idxs],
+                    "output": self.parser.pool[idx, self.parser.output_col_idxs],
+                    "pool_idxs": idx,
+                }
+            )
+        return results
+
+    def dump_results(self, base_run_dir: str) -> None:
+        """
+        Prints and saves final datasets to disk.
+
+        Args:
+            base_run_dir: Base directory path for saving.
+
+        """
+        print(100 * "=")
+        print(f"Final Results for {self.selection_method}")
+
+        metric_fname = os.path.join(base_run_dir, "final_metrics.txt")
```

Comment:
I think the full filename should be given, otherwise every time yo run with a different acquisition the file gets overwritten

Reply:
in fact, if using MLFlow or wandb or similar, better to pass the ID of the run and save that locally for ease, then use the API to figure out the metadata